packages/daemon/DESIGN.md

```diff
@@ -0,0 +1,90 @@
+
+Endo is a user agent for running guest programs.
+These programs receive capabilities from the user in the form of references to
+potentially powerful remote objects.
+To that end, Endo exhibits a system called Eventual Send that stretches
+references to remote objects over one or more capability transfer protocols
+(CapTP).
+
+A Promise represents a result from either the past or the future, and Eventual
+Send enables programs to interact with those eventual results, even if they are
+in another worker or on the far side of a network.
+Furthermore, Eventual Send allows messages to be pipelined to these eventual
+results to overcome any latency that might otherwise accrue if the program
+waited for each intermediate result before sending a follow-up message.
+
+Much hinges on what happens when a program restarts.
+Like most programs, Endo workers start without any memory from their previous
```

Comment:
Do you want to be saying "Endo workers" rather than "Endo vats"? Are these distinct concepts in Endo?

Reply:
That depends entirely on whether Endo workers meet your criteria for vats. I’m not clear whether the bar for calling a worker a “vat” is above or below orthogonal persistence, but I do know that “worker” requires no explanation and I prefer “worker” over “vat” if there isn’t a material distinction in the context of Endo. Our developer community will already know what a web worker, cloud worker, edge worker, service worker, and node worker are. Vats are certainly a species of the worker genus.

Reply:
E did manual persistence, but still had vats.

A vat is internally synchronous, with a stack and a queue, and only asynchronously coupled to other vats.

Before Shared Array Buffers, JS agents were vats. Hardened JS Agents, having (mostly) prohibited Shared Array Buffers, are vats.

Regardless, "worker" is fine. Glad to understand the distinction. Resolving.

Reply:
I’ll make a note that Endo workers qualify as vats. Thank you!